core/controllers/question_editor.py

```diff
@@ -329,12 +340,22 @@ def put(self, question_id: str) -> None:
         assert self.normalized_payload is not None
         commit_message = self.normalized_payload['commit_message']
         change_list = self.normalized_payload['change_list']
+        version = self.normalized_payload['version']
+        question = question_services.get_question_by_id(question_id)
+
+        if version > question.version:
+            raise base.BaseHandler.InvalidInputException(
+                'Trying to update version %s of question from version %s, '
+                'which is not possible. Please reload the page and try again.'
+                % (question.version, version))
 
         for change in change_list:
             if (
                     change.cmd ==
                     question_domain.CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION):
-                raise self.InvalidInputException
+                raise self.InvalidInputException(
+                    'Cannot create a new fully specified question'
+                )
 
         question_services.update_question(
             self.user_id, question_id, change_list,
```

Comment:
Can't the above check be done inside the update function, when the question is fetched? Since, the question is being fetched in the controller just for this check.

Reply:
If we do the above check in the update function, I am not able to mock the test where version of the incoming question and current question differ to write the test cases.
I believe the versioning model for question is different for that of other models of the similar type.

Reply:
Ok, so normally testing shouldn't be the main reason to choose one way of implementing things over another. 
Also, in this case, can't you pass the 'version' from the payload as an additional argument to the update question function, and do the comparison in there? This would help in mocking it as well.

Reply:
Soo should I remove validation from the controller? Cause it is done in other versioning models like topic/story.